### PR TITLE
Ignore unexpected weights from PT conversion

### LINKED ISSUE #10348: BertForMaskedLM cannot be initialized from BERT checkpoints
When I try to load a BERT model from a TF checkpoint (via `transformers-cli convert`) into a `BertForMaskedLM`, I get the following warning:
```
Some weights of BertForMaskedLM were not initialized from the model checkpoint at `SZTAKI-HLT/hubert-base-cc` and are newly initialized:
['cls.predictions.bias', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.transform.dense.weight',
'cls.predictions.transform.dense.bias', 'cls.predictions.decoder.weight', 'cls.predictions.transform.LayerNorm.weight']

You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
```

The model also performs poorly (as in: completely randomly) in masked LM. If I load a "named" model, such as `bert-base-cased`, I do not get the warning and masked LM works OK.

This is all to be expected if the tensors mentioned in the warning are indeed not part of the converted model. The question then is twofold:
1. Why aren't they? MaskedLM is one of the training tasks for a BERT model, and users rightly expect that it works (I have already received two reports for my model to that effect); i.e. that they can initialize a `BertForMaskedLM` model from a BERT checkpoint / HF model without any problems.
2. How can I convert the model so that it includes said tensors? To my knowledge, there are not options in `transformers-cli convert` that would enable me to do so.
3. The [documentation](https://huggingface.co/transformers/converting_tensorflow_models.html) should warn people of this (and better yet, describe how to convert all tensors).

## Environment info
- `transformers` version: 4.3.2
- Platform: Linux-5.4.0-60-generic-x86_64-with-debian-buster-sid
- Python version: 3.7.9
- PyTorch version (GPU?): 1.6.0 (True)
- Tensorflow version (GPU?): 2.3.0 (False)
- Using GPU in script?: N/A
- Using distributed or parallel set-up in script?: N/A

### Who can help
@LysandreJik
@sgugger

## Information

Model I am using (Bert, XLNet ...): SZTAKI-HLT/hubert-base-cc (BERT)

The problem arises when using:
* [X] the official example scripts: `transformer-cli convert`
* [ ] my own modified scripts:

The tasks I am working on is:
* [X] an official GLUE/SQUaD task: masked LM
* [ ] my own task or dataset:

## To reproduce

1. `BertForMaskedLM.from_pretrained('SZTAKI-HLT/hubert-base-cc')`
2. Observer warning messages
3. Try to use it for masked LM

## Expected behavior

Conversion: the ability to convert tensors for the training tasks
Usage: no warning messages and same MLM / NSP performance as with the official TF BERT code

### PR DESCRIPTION
**Pull request by LysandreJik**
Some weights resulting from the conversion from a PyTorch model to a TensorFlow model are throwing an unnecessary warning.

To see for yourself, the following code throws a warning before the PR:

```py
from transformers import BertForPreTraining, BertConfig, TFBertForPreTraining

pt = BertForPreTraining(BertConfig())
pt.save_pretrained("here")
tf = TFBertForPreTraining.from_pretrained("here", from_pt=True)
```

Fix https://github.com/huggingface/transformers/issues/10348

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for fixing!
